src/components/organisms/SpaceTimingForm/SpaceTimingForm.tsx

```diff
@@ -0,0 +1,219 @@
+  });
+
+  return start <= end;
+};
+
+const testEndTime: TestFunction = function testEndIsAfterStart() {
+  const { endTime, startTime, startDate, endDate } = this.parent;
+
+  const start = convertUtcSecondsFromInputDateAndTime({
+    date: startDate,
+    time: startTime,
+  });
+  const end = convertUtcSecondsFromInputDateAndTime({
+    date: endDate,
+    time: endTime,
+  });
+
+  return start < end;
+};
+
+const validationSchema = Yup.object().shape({
+  startDate: Yup.string()
+    .required()
+    .test(
+      "startDate",
+      "Start date must not be after the end date",
+      testEndDate
+    ),
+  startTime: Yup.string()
+    .required()
+    .test("startTime", "Start time must be before the end time", testEndTime),
+  endDate: Yup.string()
+    .required()
+    .test(
+      "endDate",
+      "End date must not be before the starting date",
+      testEndDate
+    ),
+  endTime: Yup.string()
+    .required()
+    .test("endTime", "End time must be after the start time", testEndTime),
+});
+
+export interface SpaceTimingFormProps extends AdminSidebarFooterProps {
+  venue?: WithId<AnyVenue>;
+}
+
+export const SpaceTimingForm: React.FC<SpaceTimingFormProps> = ({ venue }) => {
+  const { user } = useUser();
+
+  const defaultValues = useMemo<SpaceTimingFormInput>(() => {
+    const {
+      inputFormattedDateSegment: startDate,
+      inputFormattedTimeSegment: startTime,
+    } = convertDateFromUtcSeconds(venue?.start_utc_seconds ?? NaN);
+
+    const {
+      inputFormattedDateSegment: endDate,
+      inputFormattedTimeSegment: endTime,
+    } = convertDateFromUtcSeconds(venue?.end_utc_seconds ?? NaN);
+
+    return {
+      startTime,
+      startDate,
+      endTime,
+      endDate,
+    };
+  }, [venue?.start_utc_seconds, venue?.end_utc_seconds]);
+
+  const {
+    reset,
+    formState: { dirty, isSubmitting },
+    register,
+    errors,
+    handleSubmit,
+  } = useForm<SpaceTimingFormInput>({
+    mode: "onSubmit",
+    reValidateMode: "onChange",
+    validationSchema,
+    defaultValues,
+  });
+
+  const [{ error, loading: isSaving }, submit] = useAsyncFn(
+    async (input: SpaceTimingFormInput) => {
+      if (!venue?.name || !user) return;
+
+      await updateVenue_v2(
+        {
+          name: venue?.name,
+          worldId: venue?.worldId,
+          start_utc_seconds: convertUtcSecondsFromInputDateAndTime({
+            date: input.startDate,
+            time: input.startTime,
+          }),
+          end_utc_seconds: convertUtcSecondsFromInputDateAndTime({
+            date: input.endDate,
+            time: input.endTime,
+          }),
+        },
+        user
+      );
+
+      reset(input);
+    },
+    [venue, user, reset]
+  );
+
+  const isSaveLoading = isSubmitting || isSaving;
+  const isSaveDisabled = !(
+    // Useful if form has mode: "onChange"
+    // Object.keys(errors).length ||
+    (dirty || isSaving || isSubmitting)
+  );
+
+  if (!venue) {
+    return <LoadingPage />;
```

Comment:
Another potential infinite loading?

Reply:
Probably the same comment as in https://github.com/sparkletown/sparkle/pull/2512#discussion_r741922588 applies, but in the form case it shouldn't be a loading page in the first place, and might probably also work with `venue?.` instead of just not rendering anything.

Reply:
Deleted the `if` @ c2bf6c9a13bd4ed9cafc26a2143a2df5a05ab383